plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/RegisterTableProcedure.java

```diff
@@ -0,0 +1,134 @@
+            "Valid usages:%n" +
+                    " - %1$s(%2$s => ..., %3$s => ..., %4$s => ...)%n" +
+                    " - %1$s(%2$s => ..., %3$s => ..., %4$s => ..., %5$s => ...)",
+            PROCEDURE_NAME,
+            // Use lowercase parameter names per convention. In the usage example the names are not delimited.
+            SCHEMA_NAME.toLowerCase(ENGLISH),
+            TABLE_NAME.toLowerCase(ENGLISH),
+            TABLE_LOCATION.toLowerCase(ENGLISH),
+            METADATA_FILE_NAME.toLowerCase(ENGLISH));
+
+    static {
+        try {
+            REGISTER_TABLE = lookup().unreflect(RegisterTableProcedure.class.getMethod("registerTable", ConnectorSession.class, String.class, String.class, String.class, String.class));
+        }
+        catch (ReflectiveOperationException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    private final TrinoCatalogFactory catalogFactory;
+    private final ClassLoader classLoader;
+
+    @Inject
+    public RegisterTableProcedure(TrinoCatalogFactory catalogFactory)
+    {
+        this.catalogFactory = requireNonNull(catalogFactory, "catalogFactory is null");
+        // this class is loaded by PluginClassLoader, and we need its reference to be stored
+        this.classLoader = getClass().getClassLoader();
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                SYSTEM_SCHEMA,
+                PROCEDURE_NAME,
+                ImmutableList.of(
+                        new Procedure.Argument(SCHEMA_NAME, VARCHAR),
+                        new Procedure.Argument(TABLE_NAME, VARCHAR),
+                        new Procedure.Argument(TABLE_LOCATION, VARCHAR),
+                        new Procedure.Argument(METADATA_FILE_NAME, VARCHAR, false, null)),
+                REGISTER_TABLE.bindTo(this));
+    }
+
+    public void registerTable(ConnectorSession clientSession, String schemaName, String tableName, String tableLocation, String metadataFileName)
+    {
+        // this line guarantees that classLoader that we stored in the field will be used inside try/catch
+        // as we captured reference to PluginClassLoader during initialization of this class
+        // we can use it now to correctly execute the procedure
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
+            doRegisterTable(
+                    clientSession,
+                    schemaName,
+                    tableName,
+                    tableLocation,
+                    Optional.ofNullable(metadataFileName));
+        }
+    }
+
+    private void doRegisterTable(ConnectorSession clientSession, String schemaName, String tableName, String tableLocation, Optional<String> metadataFileName)
+    {
+        if (schemaName == null || tableName == null || tableLocation == null) {
+            throw new TrinoException(INVALID_ARGUMENTS, "Illegal parameter set passed. " + PROCEDURE_USAGE_EXAMPLES);
```

Comment:
See https://github.com/trinodb/trino/blob/e4e0d197f4fe2e7889586a1503aaf7692601fbd4/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/DropExtendedStatsProcedure.java#L77-L78 for a better example on how to validate procedure args.

Reply:
Addressed!